### PR TITLE
Defensive on null arguments

### PR DESCRIPTION
**Pull request by Corniel**
There was no null check involved in the dynamic dispatcher. While visiting the code, some tests/specs where rewritten.

### CODE REVIEW

File: test/Qowaiv.DomainModel.UnitTests/Dynamic/DynamicEventDispatcher_specs.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
+﻿using Microsoft.CSharp.RuntimeBinder;
+using NUnit.Framework;
+using Qowaiv.DomainModel.Dynamic;
+using System;
+
+namespace Dynamic_event_dispatcher_specs
+{
+    public class Support_when_mehods
```
Comment (LauraXiulan) [inline test/Qowaiv.DomainModel.UnitTests/Dynamic/DynamicEventDispatcher_specs.cs:8]: methods

File: test/Qowaiv.DomainModel.UnitTests/Dynamic/DynamicEventDispatcher_specs.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    {
+        [Test]
+        public void With_single_argument() => AssertEventTypes<WhenWithSingleArgument>();
+        [Test]
+        public void With_return_statement() => AssertEventTypes<WhenWithSingleArgument>();
```
Comment (LauraXiulan) [inline test/Qowaiv.DomainModel.UnitTests/Dynamic/DynamicEventDispatcher_specs.cs:13]: WhenWithReturnStatement

**Review by LauraXiulan — APPROVED** [review]

File: test/Qowaiv.DomainModel.UnitTests/Dynamic/DynamicEventDispatcher_specs.cs (hunk 2 above)
Comment (Corniel) [reply test/Qowaiv.DomainModel.UnitTests/Dynamic/DynamicEventDispatcher_specs.cs]: Nice catch!

File: test/Qowaiv.DomainModel.UnitTests/Dynamic/DynamicEventDispatcher_specs.cs [hunk 3]
```diff
@@ -5,12 +5,12 @@
 
 namespace Dynamic_event_dispatcher_specs
 {
-    public class Support_when_mehods
+    public class Support_when_methods
     {
         [Test]
-        public void With_single_argument() => AssertEventTypes<WhenWithSingleArgument>();
+        public void With_single_argument() => AssertEventTypes<WhenWithReturnStatement>();
```
Comment (LauraXiulan) [inline test/Qowaiv.DomainModel.UnitTests/Dynamic/DynamicEventDispatcher_specs.cs:11]: This one didn't need change ;)

Comment (Corniel) [reply test/Qowaiv.DomainModel.UnitTests/Dynamic/DynamicEventDispatcher_specs.cs:11]: Oops!

**Pull request merged**